### DISCUSSION TITLE
Make live code editable in MyST Website

### DISCUSSION THREAD

**Discussion by nikolaimatni** [executablebooks/meta#1172] [Q&A]
Hello,

Is there a configuration setting that makes executable code in MyST websites turn into an editor, for example, as is shown here: https://executablebooks.github.io/thebe/

If working with Jupyter Book, this happens by default if I set `thebe: true` in `_config.yml`.  However, when working with a [MyST website](https://github.com/executablebooks/mystmd-quickstart), while setting `jupiter: true` in `myst.yml` lets you connect to binder and gives you buttons to run the code in a notebook, you can't actually edit it live.

Thanks!
Nik

**Comment by choldgraf** [comment]
There is support for thebe with mystmd though it is still being developed. Have you tried the instructions on the [integrating computation with jupyter](https://mystmd.org/guide/integrating-jupyter)  docs?

Cc also @stevejpurves who I believe has worked on this the most and may have pointers.

**Reply by nikolaimatni** [reply]
Thanks for your answer!  Yes, those are exactly the instructions that I followed: they were great in making the notebooks/codeblocks executable. What I haven't been able to suss out is what the appropriate option/setting is to make the code blocks editable as well.  

If the output were static html, the thebe documentation explains that marking a code snippet as `<‍pre data-executable="true" data-language="python"><‍/pre>` is what's needed to make thebe will turn it into an editor when activated.  I'm looking for the analogous command to insert into myst.yml or the .json output by the mystmd build.

**Reply by stevejpurves** [reply]
Hi @nikolaimatni,

yes, those `thebe` options are when you are working with a static html site and adding in `thebe` functionality. For this to work in the myst themes we need to do more work to enable this, drop in codemirror and do a state management upgrade along the way. So it's not currently supported, it will be I am sure but I'm not sure when we'll be doing that work. The requests for this feature are growing though!

**Reply by nikolaimatni** [reply]
Appreciate the explanation, and will keep an eye out for any updates, thanks!
